File: Python_Assignment/Python_Programming_Assignment_2/Assignment2_3.py
# 3. Write a program which accept one number from user and return its factorial.

# Input : 5                 Output : 120

# To find factorial
def Fact(No):
    if No == 0:
        print("Factorial of 0 is : 1")
        return

    # checking if given number is negative
    if No < 0:
        print("Enter valied input, non-negative numbers only")
        return

    fact = 1
    i = 1
    while(i<=No):
        fact *= i
        i += 1
    
   
    print("Factorial of ", No, " is : ", fact)
    
def main():
    # Taking input from user
    no = int(input("Please enter number to find Factorial : "))
    Fact(no)
    
if __name__ == "__main__":
    main()

File: Typescript_Assignment/Typescript_Assignment2/Assignment2_3.ts
// 2.3 Write a typescript program which contains one function named as maximum.
//     That function accepts array of numbers and returns the second largest
//     number from array.

// Input : 23 89 6 29 56 45 77 32
// Output : Second Maximum number is 77.

function MaxiMum(Data : number[]) : number
{
    var max : number = Data[0]
    var smax : number = Data[0]
    
    for(var i = 0; i<Data.length; i++)
    {
        if(max<=Data[i])
        {
            smax = max
            max = Data[i]
        }
        else if(smax<=Data[i] && smax!=max)
        {
            smax = Data[i]
        }
    }
    return smax
}

var Data : number[] = new Array()
Data = [23, 89, 6, 29, 56, 77, 32]

console.log("Second Maximum number is :"+MaxiMum(Data))